trace/v2/config.go

```diff
@@ -2,58 +2,92 @@ package trace
 
 import (
 	"errors"
+	"fmt"
 	"strings"
 
 	"github.com/arquivei/foundationkit/app"
 	"github.com/rs/zerolog/log"
 	"go.opentelemetry.io/otel"
 	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
 	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.21.0"
 )
 
 // Config represents all trace's configuration
 type Config struct {
-	Exporter          string  `default:""`
-	ProbabilitySample float64 `default:"0"`
+	ServiceName       string
+	ServiceVersion    string
+	Exporter          string
+	ProbabilitySample float64
 	Stackdriver       StackdriverConfig
 	OTLP              OTLPConfig
 }
 
 // Setup use Config to setup an trace exporter and returns a shutdown handler
 func Setup(c Config) app.ShutdownFunc {
+	exporter, err := newExporter(c)
+
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to create trace exporter")
+	}
+
+	res, err := newResource(c.ServiceName, c.ServiceVersion)
+	if err != nil {
+		log.Fatal().Err(err).Msg("Failed to create trace resource")
+	}
+
+	tp := newTraceProvider(res, c.ProbabilitySample, exporter)
+
+	otel.SetTracerProvider(tp)
+	otel.SetTextMapPropagator(newPropagator())
+
+	return tp.ForceFlush
+}
+
+func newResource(serviceName, serviceVersion string) (*resource.Resource, error) {
+	return resource.Merge(resource.Default(),
+		resource.NewWithAttributes(semconv.SchemaURL,
+			semconv.ServiceName(serviceName),
```

Comment:
Are `serviceName` and `serviceVersion` required? If they are, does `resource` return a good error message? If not, a custom error message would make it quick to someone who is using fkit without deep knowledge of this source code to know which configurations are missing